packages/typespec-client-generator-core/src/package.ts

```diff
@@ -463,11 +424,18 @@ function getSdkMethods<TServiceOperation extends SdkServiceOperation>(
     const operationGroupClient = diagnostics.pipe(
       createSdkClientType<TServiceOperation>(context, operationGroup, sdkClientType),
     );
-    const clientInitialization = getClientInitialization(context, operationGroup.type);
+    const tspClientInitialization = getClientInitialization(context, operationGroup.type);
     const parameters: SdkParameter[] = [];
-    if (clientInitialization) {
-      for (const property of clientInitialization.properties) {
-        parameters.push(property);
+    if (tspClientInitialization) {
+      const clientInitialization = getSdkInitializationType(
+        context,
+        operationGroup,
+        tspClientInitialization,
+      );
+      for (const property of clientInitialization.model.properties) {
+        if (property.kind === "method") {
```

Comment:
is there any case the initialization model property is not a parameter?

Reply:
no, i'm making them all method parameters because we also use that in the logic to pull them out of the method signature